harness/determined/pytorch/_pytorch_context.py

```diff
@@ -51,9 +51,18 @@ class PyTorchTrialContext(det.TrialContext, pytorch._PyTorchReducerContext):
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         det.TrialContext.__init__(self, *args, **kwargs)
         pytorch._PyTorchReducerContext.__init__(self, self.distributed.allgather)
+
+        self.cluster_info = det.get_cluster_info()
+        if self.cluster_info:
+            self.managed_training = True
+        else:
+            self.managed_training = False
```

Comment:
Is this accurate?  What if I want to do local training, but I happen to be in a determined notebook or shell?